packages/snaps-sdk/src/structs.ts

```diff
@@ -0,0 +1,21 @@
+import type { Struct } from '@metamask/superstruct';
+import { record, refine } from '@metamask/superstruct';
+
+/**
+ * Refine a struct to be a non-empty record.
+ *
+ * @param Key - The struct for the record key.
+ * @param Value - The struct for the record value.
+ * @returns The refined struct.
+ */
+export function nonEmptyRecord<Key extends string, Value>(
+  Key: Struct<Key>,
+  Value: Struct<Value>,
+) {
+  return refine(record(Key, Value), 'Non-empty record', (value) => {
+    return (
+      (Array.isArray(value) && value.length > 0) ||
```

Comment:
Wait I thought we didn't want to allow arrays here

Reply:
IMO we shouldn't break the existing assumption that arrays can be used if someone is already doing it and keep record as close as the JS definition is ?

Reply:
at least this checks that the array is not empty

Reply:
I think we would want to restrict it further because during JSON serialization properties on an array aren't included

Reply:
Okay, then disallow arrays :)

Reply:
IMO we can discuss for each case when we start using `nonEmptyRecord` whether it would be breaking. My general thought is that it may be better to restrict everywhere to prevent unexpected issues

Reply:
That's true and if we pass everything through JSON-RPC it make sence to disallow it